src/HttpClient/GuzzleHttpClient.php

```diff
@@ -31,16 +32,16 @@ class GuzzleHttpClient implements HttpClientContract
     /**
      * GuzzleHttp client.
      *
-     * @var \GuzzleHttp\Client
+     * @var \GuzzleHttp\ClientInterface
      */
     protected $httpClient;
 
     /**
      * Create a new GuzzleHttpClient instance.
      *
-     * @param \GuzzleHttp\Client|null $httpClient
+     * @param \GuzzleHttp\ClientInterface $httpClient
      */
-    public function __construct(Client $httpClient = null)
+    public function __construct(ClientInterface $httpClient)
```

Comment:
Replacing concrete class `Client` with it's interface not breaking because it's done in constructor and not in method.

And default `null` value was a bug, because this class can't work when `$httpClient` is `null`.